src/utils/i18n-utils.js

```diff
@@ -25,16 +25,34 @@ export function localeIsValid(localeCode) {
 
 /**
  * Updates i18n global var and html lang
- * @param {{ params: { locale: String } }} to - where the route navigates to
+ * @param {String} locale - locale used
  * @param {Object{}} env - context
- * @param {{ code: String, name: String }[]} locales
  */
-export function updateCurrentLocale(to, env) {
-  const currentLocale = to.params.locale || defaultLocale;
+export function updateLocale(locale = defaultLocale, env) {
   // exist if current locale is not supported
-  if (!localeIsValid(currentLocale)) return;
+  if (!localeIsValid(locale)) return;
   // update locale global var
-  env.$i18n.locale = currentLocale; // eslint-disable-line no-param-reassign
+  env.$i18n.locale = locale; // eslint-disable-line no-param-reassign
   // update html lang
-  updateLangTag(currentLocale);
+  updateLangTag(locale);
+}
+
+/**
+ * Updates i18n global var and html lang
+ * @param {{ params: { locale: String } }} to - where the route navigates to
+ * @param {Object{}} env - context
+ */
+export function updateCurrentLocale(to, env) {
+  const currentLocale = to.params.locale;
+  updateLocale(currentLocale, env);
+}
+
+/**
+ * Get locale slug from locale code
+ * @param {String} localeCode - locale code
+ * @return {String}
+ */
+export function getSlug(localeCode) {
```

Comment:
Not really blocking feedback but an optimization of this to avoid doing a search with `find` every time this is called would be to create a mapping at the top level so this function is always a constant lookup.

Something like:
```js
const slugForLocale = locales.reduce((map, locale) => ({
  ...map,
  [locale.slug]: locale.code,
}), {});

// ...

export function getSlugForLocale(localeCode) {
  return slugForLocale(localeCode);
}
```

Reply:
(Or the reverse if you end up doing the reverse mapping for `getLocaleCode(localeSlug)` instead of course)

Reply:
great optimization! thanks a lot :)